### PR TITLE
feature: Filtering conversations and contacts with custom attributes

### LINKED ISSUE #3738: Support the UI to have Custom Attributes as filters
- Add a new section in the filters dropdown with the list of supported custom attributes.

eg: 
Screenshot 2022-01-11 at 7 39 01 PM

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description

Adds support for filtering conversations and contacts with custom attributes

Fixes #3738 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue [hunk 1]
```diff
@@ -58,10 +59,11 @@
 <script>
 import alertMixin from 'shared/mixins/alertMixin';
 import { required } from 'vuelidate/lib/validators';
-import FilterInputBox from '../../../../components/widgets/FilterInput.vue';
+import FilterInputBox from '../../../../components/widgets/FilterInput/Index.vue';
 import countries from '/app/javascript/shared/constants/countries.js';
 import { mapGetters } from 'vuex';
-
+import { filterAttributeGroups } from '../contactFilterItems';
+import * as OPERATORS from '../../../../components/widgets/FilterInput/FilterOperatorTypes';
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue:66]: Use absolute path here

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/components/widgets/conversation/ConversationAdvancedFilter.vue [hunk 2]
```diff
@@ -48,22 +49,24 @@
...
 import { required, requiredIf } from 'vuelidate/lib/validators';
-import FilterInputBox from '../FilterInput.vue';
+import FilterInputBox from '../FilterInput/Index.vue';
 import languages from './advancedFilterItems/languages';
 import countries from '/app/javascript/shared/constants/countries.js';
 import { mapGetters } from 'vuex';
-
+import { filterAttributeGroups } from './advancedFilterItems';
+import filterMixin from 'shared/mixins/filterMixin';
+import * as OPERATORS from '/app/javascript/dashboard/components/widgets/FilterInput/FilterOperatorTypes.js';
```
Comment (nithindavid) [inline app/javascript/dashboard/components/widgets/conversation/ConversationAdvancedFilter.vue:58]:
```suggestion
import * as OPERATORS from 'dashboard/components/widgets/FilterInput/FilterOperatorTypes';
```

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue [hunk 3]
```diff
@@ -58,21 +59,23 @@
...
 import alertMixin from 'shared/mixins/alertMixin';
 import { required } from 'vuelidate/lib/validators';
-import FilterInputBox from '../../../../components/widgets/FilterInput.vue';
+import FilterInputBox from '../../../../components/widgets/FilterInput/Index.vue';
 import countries from '/app/javascript/shared/constants/countries.js';
 import { mapGetters } from 'vuex';
-
+import { filterAttributeGroups } from '../contactFilterItems';
+import filterMixin from 'shared/mixins/filterMixin';
+import * as OPERATORS from '/app/javascript/dashboard/components/widgets/FilterInput/FilterOperatorTypes.js';
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue:67]:
```suggestion
import * as OPERATORS from 'dashboard/components/widgets/FilterInput/FilterOperatorTypes.js';
```

File: app/javascript/dashboard/routes/dashboard/contacts/contactFilterItems/index.js [hunk 4]
```diff
@@ -1,137 +1,88 @@
+import {
+  OPERATOR_TYPES_1,
+  OPERATOR_TYPES_3,
+} from '/app/javascript/dashboard/components/widgets/FilterInput/FilterOperatorTypes.js';
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/contactFilterItems/index.js:4]:
```suggestion
} from 'dashboard/components/widgets/FilterInput/FilterOperatorTypes.js';
```

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Review by nithindavid — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]
QA has been done on local environment and everything is working as expected.

**Pull request merged**